dart/dynamics/Entity.h

```diff
@@ -189,6 +198,19 @@ class Entity : public virtual common::Subject
   bool needsAccelerationUpdate() const;
 
 protected:
+
+  /// Used when constructing a Frame class, because the Frame constructor will
+  /// take care of setting up the parameters you pass into it
+  enum ConstructFrame_t { ConstructFrame };
+
+  explicit Entity(ConstructFrame_t);
```

Comment:
That's a very clever use of type tags! It's a shame this is necessary, but this is a very nice solution. :+1:

Type tags are also found in the C++11 standard library, including the constructor of `std::unique_lock`.